scrapers/lobbyist/scrape_lobbyists.py

```diff
@@ -1,90 +1,103 @@
-        "MemberID",
-        "Phone",
-        "MemberVersionID",
-        "Year",
-        "LobbyingEfforts",
-        "LobbyistType",
-        "LobbyistName",
-        "LobbyistAddress",
-        "LobbyingFirm",
-        "PrincipalContact",
-        "EmploymentDate",
-        "EmailAddress",
-        "TotalRows",
-        "RowNumber",
-        "ReportFileName",
-        "RegistrationID",
-        "IsPastRegistrationExists",
-        "Email",
-        "CompensatedStatus",
-        "LobbyingSourceFunds",
-        "AddressLines",
-        "City",
-        "StateCode",
-        "CountryCode",
-        "ZipCode",
-        "CityAddress",
-        "CreatedBy",
-        "CreatedDate",
-        "CreatedIPAddress",
-        "LobbyistClientID",
-        "LobbyistClientVersionID",
-        "LastModifiedBy",
-        "LastModifiedDate",
-        "LastModifiedIPAddress",
-        "Message",
-        "IsSuccess",
-        "Action",
-        "IsDirty",
-        "PersonID",
-        "PersonVersionID",
-        "ClientID",
-    ],
-)
-
-writer.writeheader()
-
-s = scrapelib.Scraper(requests_per_minute=0, retry_attempts=3, verify=False)
-
-payload = {
-    "PageNo": 1,
-    "PageSize": 1000,
-    "ElectionYear": "",
-    "SortDir": "ASC",
-    "SortedBy": "",
-    "ClientVersionID": "",
-}
-
-for row in tqdm(reader):
-    _payload = payload.copy()
-    _payload["ClientID"] = row["ClientID"]
-    _payload["ClientVersionID"] = str(int(float(row["ClientVersionID"])))
-
-    response = s.post(
-        "https://login.cfis.sos.state.nm.us/api//ClientDetails/LobbyistsByClientList",
-        data=json.dumps(_payload),
-        headers={"Content-Type": "application/json"},
+logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
+
+
+def main(rpm=180, retries=3, verify=False):
+    writer = csv.DictWriter(
+        sys.stdout,
+        fieldnames=[
+            "LobbyistName",
+            "ID",
+            "MemberVersionID",
+            "Clients",
+            "Email",
+            "AddressLines",
+            "CityAddress",
+            "LobbyPhone",
+        ],
+        extrasaction="ignore",
+    )
+
+    writer.writeheader()
+
+    s = scrapelib.Scraper(
+        requests_per_minute=rpm, retry_attempts=retries, verify=verify
     )
 
-    if response.ok:
-        for record in response.json():
-            record["ClientID"] = row["ClientID"]
-            writer.writerow(record)
+    payload = {
+        "ElectionYear": "",
+        "Status": "",
+        "TransactionType": None,
+        "TransactionAmount": 0,
+        "NumberofClients": None,
+        "IsCompliance": None,
+        "TotalCompensation": None,
+        "LobbyistType": "",
+        "NumberofLobbyists": None,
+        "ClientBusinessorEntityType": None,
+        "AgentType": "",
+        "PageNo": 1,
+        "PageSize": 1000,
+        "SortOrder": "ASC",
+    }
+
+    page_number = 1
+    page_size = 1000
+    result_count = 1000
+    seen_lobbyists = deque(maxlen=25)
```

Comment:
Ditto deduplicated. In addition, our previous approach for scraping lobbyist employers -> scraping lobbyists no longer works????? If you go to the lobbyist tab for an employer with multiple lobbyists and click around the list, you'll notice the lobbyist IDs are not correct and the links don't resolve correctly. So, I updated this script to scrape the lobbyist search, and for each distinct lobbyist, retrieve details. I scraped their employers (clients) separately.